syntax="proto3";
package market_communication_with_seller;
service MarketPlace
{
    rpc RegisterSeller(RegisterSellerRequest) returns (StatusResponse);

    rpc SellItem(SellItemRequest) returns (StatusResponse);

    rpc UpdateItem(UpdateItemRequest) returns (StatusResponse);

    rpc DeleteItem(DeleteItemRequest) returns (StatusResponse);
    
    rpc DisplaySellerItems(ProductDisplayRequest) returns (stream ProductDisplayResponse);
}
message RegisterSellerRequest
{
    string address = 1;
    string uuid = 2;
}
enum Category
{
    ELECTRONICS=0;
    FASHION=1;
    OTHERS=2;
    ANY=3;
}
message SellItemRequest
{
    string productName = 1;  
    Category productCategory = 2;
    uint32 quantity = 3;
    string description=4;
    float pricePerUnit = 5;
    string uuid=6;
}
message UpdateItemRequest
{
    uint64 id=1;
    float newPrice=2;
    uint32 newQuantity=3;
    string uuid=4;
}
message DeleteItemRequest
{
    uint64 id=1;
    string uuid=2;
}
message ProductDisplayRequest
{
    string uuid = 1;
}
message ProductDisplayResponse
{
    uint64 id = 1;
    float price = 2;
    string name = 3;
    Category productCategory=4;
    uint32 quantityRemaining=5;
    float rating = 6;
    string sellerAddress=7;
    string description = 8;
}
message StatusResponse
{
    string status=1;
}